doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -7,64 +7,179 @@ In this chapter, we introduce advanced commands to modify the way Coq
 parses and prints objects, i.e. the translations between the concrete
 and internal representations of terms and commands.
 
-The main commands to provide custom symbolic notations for terms are
-:cmd:`Notation` and :cmd:`Infix`; they will be described in the
-:ref:`next section <Notations>`. There is also a
-variant of :cmd:`Notation` which does not modify the parser; this provides a
-form of :ref:`abbreviation <Abbreviations>`. It is
-sometimes expected that the same symbolic notation has different meanings in
-different contexts; to achieve this form of overloading, Coq offers a notion
-of :ref:`notation scopes <Scopes>`.
-The main command to provide custom notations for tactics is :cmd:`Tactic Notation`.
+When inputing a term to Coq, it goes through several successive steps.
```

Comment:
This makes a lot of sense. However I don't know what's the best way of implementing it here. In particular since the explanation is needed to understand the following such paragraph (even before this PR it was already mentioning that abbreviation "does not modify the parser"). What should we do
* add yet another such introductory paragraph, about what notations can be used for, before the explanation
* just permute the explanation and the following "sectioning" paragraph, even if it means getting the definitions after their first use
* something else?

Reply:
Thanks, I did try that.